### PR TITLE
RISC-V: Add Makefile variables for platform ISA, ABI, and code model

### PR DESCRIPTION
**Pull request by gagachang**
In RISC-V, each platform may have different supported ISA extensions, ABI, and code model.

In the first commit, we define the default variables of ISA extensions, ABI, and code model in RISC-V core Makefile `riscv.mk`.
In the second commit, we show that how the virt platform overrides the ISA in its `conf.mk`.

### CODE REVIEW

File: core/arch/riscv/riscv.mk [hunk 1]
```diff
@@ -71,9 +71,20 @@ core-platform-cppflags	+= -I$(arch-dir)/include
...
+rv64-platform-abi ?= lp64d
+rv32-platform-isa ?= rv32imafd
+rv32-platform-abi ?= ilp32d
+
+rv64-platform-cppflags += -mcmodel=$(riscv-platform-mcmodel)
+rv64-platform-cppflags += -march=$(rv64-platform-isa) -mabi=$(rv64-platform-abi)
 rv64-platform-cppflags += -Wno-missing-include-dirs
+rv32-platform-cppflags += -mcmodel=$(riscv-platform-mcmodel)
+rv32-platform-cppflags += -march=$(rv32-platform-isa) -mabi=$(rv32-platform-abi)
+rv32-platform-cppflags += -Wno-missing-include-dirs
```
Comment (jenswi-linaro) [inline core/arch/riscv/riscv.mk:87]: Why is this needed?

Comment (gagachang) [reply core/arch/riscv/riscv.mk]: @jenswi-linaro Resolved in third commit.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by gagachang** [comment]
Tag applied!
